### PR TITLE
feat: teach *Metadata and ScalarValue to Display

### PR DESCRIPTION
**Pull request by danking**
`tree_display` is a fantastic debugging tool until you get a ConstantArray whose value is a BufferString of a Buffer which is a 1 character slice of a buffer of 65,536 "F"s which get Debug'ed as the number 70 with a comma and a space repeated 65,536 times.

### CODE REVIEW

**Comment by robert3005** [comment]
so if you want to delegate in rust you could just do `Debug::fmt(self, f)`

**Comment by danking** [comment]
Done.

File: vortex-array/src/tree.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ impl<'a, 'b: 'a> ArrayVisitor for TreeFormatter<'a, 'b> {
                 writeln!(
                     i.fmt,
                     // TODO(ngates): use Display for metadata
```
Comment (robert3005) [inline vortex-array/src/tree.rs]: Kill this comment

**Review by robert3005 — APPROVED** [review]

**Comment by robert3005** [comment]
I made a quick commit on top here to remove `#[allow(clippy::use_debug)]` as well

**Comment by robert3005** [comment]
FWIW I think we can now simplify display for scalar since it does similar logic

**Pull request merged**